pandas/core/generic.py

```diff
@@ -635,6 +635,86 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
         d[axis] = mapper
         return self.rename(**d)
 
+    def set_axis_name(self, name=None, level_names=None, axis=0,
```

Comment:
I would suggest parsing the first argument as a list of names if the index is a MultiIndex.